src/librustdoc/rustdoc-proc-macro/src/lib.rs

```diff
@@ -0,0 +1,57 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn;
+
+#[proc_macro_derive(ItemTemplate)]
+pub fn item_template_derive(input: TokenStream) -> TokenStream {
+    let ast = syn::parse::<syn::DeriveInput>(input).unwrap();
+    let name = &ast.ident;
+    let impl_disp_capt_a_b_capt_cx = quote! {
+        impl std::fmt::Display + rustc_data_structures::captures::Captures<'a> + 'b + rustc_data_structures::captures::Captures<'cx>
+    };
+    let gen = quote! {
+        impl<'a, 'cx: 'a> ItemTemplate<'a, 'cx> for #name<'a, 'cx> {
+            fn item_and_mut_cx(&self) -> (&'a crate::clean::Item, std::cell::RefMut<'_, &'a mut crate::html::render::Context<'cx>>) {
+                (self.it, self.cx.borrow_mut())
```

Comment:
Would be nice to check if the type actually has these fields and generate a `compile_error!()` (with the type span) if it doesn't.